src/libexpr/primops/fetchTree.cc

```diff
@@ -350,6 +351,8 @@ static RegisterPrimOp primop_fetchTree({
           Number of revisions in the history of the Git repository before the fetched commit.
 
           If set, pass through the value to the output attribute set.
+          The provided value will be checked against the one obtained from the repository, and an error is thrown if there is a mismatch.
```

Comment:
I don't think this is correct. AFAIK we don't do such a check.

Reply:
@DavHau says we do: https://github.com/NixOS/nix/pull/9258#issuecomment-1870252616